LEAF_Nexus/auth_cookie/index.php

```diff
@@ -19,14 +19,21 @@
 if (isset($_COOKIE['REMOTE_USER']))
 {
     $redirect = '';
-    if (isset($_GET['r']))
-    {
-        $redirect = $protocol . HTTP_HOST . base64_decode($_GET['r']);
-    }
-    else
-    {
+
+    if (isset($_GET['r'])) {
+        $encodedRedirect = $_GET['r'];
+        $decodedRedirect = base64_decode($encodedRedirect);
+    
+        if ($decodedRedirect !== false) {
+            $parsedRedirect = parse_url($decodedRedirect);
+    
+            if ($parsedRedirect !== false) {
+                $redirect = $protocol . HTTP_HOST . $parsedRedirect['path'];
+            }
+        }
+    } else {
         $redirect = $protocol . HTTP_HOST . dirname($_SERVER['PHP_SELF']) . '/../';
```

Comment:
This should be used to initialize $redirect as the default. This would avoid having a blank redirect due to the new nested conditions.